### ISSUE TITLE
Buffered channels output differs 

### ISSUE THREAD

**Issue by goksan** [crystal-lang/crystal-book#536]
Hi there 👋, Crystal noob here getting stuck in this weekend 

I'm confused by the [Concurrency > Buffered Channels](https://crystal-lang.org/reference/guides/concurrency.html#buffered-channels) section having gotten a different output from the snippet copied verbatim.

Crystal version:
```
crystal -v
Crystal 1.1.1 [6d9a1d583] (2021-07-26)

LLVM: 10.0.1
Default target: x86_64-unknown-linux-gnu
```

Output in docs:
```
Before send 1
Before send 2
Before send 3
1
2
After send
3
```

Output I got:
```
crystal test.cr
Before send 1
Before send 2
Before send 3
After send
1
2
3
```

I've also tried on: https://play.crystal-lang.org/#/r/bsr4 where I got the same output which differs from the docs

**Comment by goksan** [comment]
From what I can surmise this has been incorrect from versions later than `0.30.1` having played around with the compiler version in the playground

**Comment by Blacksmoke16** [comment]
Related: https://github.com/crystal-lang/crystal/issues/10977

**Comment by straight-shoota** [comment]
Resolved by #537
